openqa-investigate

```diff
@@ -137,7 +137,13 @@ investigate() {
         [[ ! "$old_name" =~ investigate:retry$ ]] && echo "Job is ':investigate:' already, skipping investigation" && return 0
         investigate_origin="$(echo "$job_data" | runjq -r '.job.settings.OPENQA_INVESTIGATE_ORIGIN')" || return 1
         origin_job_id=${investigate_origin#$host_url/t}
-        "${client_call[@]}" -X POST jobs/"$origin_job_id"/comments text="Investigate retry job: $host_url/t$id failed, likely not a sporadic test"
+        out=$("${client_call[@]}" -X POST jobs/"$origin_job_id"/comments text="Investigate retry job: $host_url/t$id failed, likely not a sporadic test" 2>/dev/null)
+        # meanwhile the original job might have been deleted already, handle
+        # gracefully
+        if [[ $(echo "$out" | runjq .error_status) != "404" ]]; then
+            echo "Unexpected error encountered when posting comments: '$out'"
+            exit 2
```

Comment:
The `investigate` function is run in a loop for multiple user-specified, independent jobs. Shouldn't the script better continue with the next job instead of terminating completely?

Reply:
you are right. As we already count the errors I turned this into `return 2`